main.go

```diff
@@ -38,6 +58,7 @@ func authorize(w http.ResponseWriter, r *http.Request) {
 
 	tmpl := template.Must(template.ParseFiles("static/index.html"))
 	data := AuthorizePage{
+		SelfRoot:   SelfRoot(r),
 		Authorized: true,
 		URL:        url,
```

Comment:
`url` (defined above) is still being populated by the `REDIRECT_URI` env var, so when a user attempts to authorize the against a Plaxt instance that only has `ALLOWED_HOSTNAME` they won't get a usable URL. 

I think you can just drop `SelfRoot(r)` in the place of the `Getenv` and everything will be groovy.

Reply:
:P i had to re-read that like 3 times. I made a wrong assumption of what you were saying. You are saying get rid of the os.Getenv from the `url :=` declaration, not drop SelfRoot from the struct.

Yea! i can do it

Once i figure out more about how interfaces work, i'll see if i can't make more tests for the handlers (cause of trakt.AuthRequest).